lib/project_types/extension/features/argo.rb

```diff
@@ -30,7 +30,12 @@ class Argo
       ].freeze
 
       def create(directory_name, identifier, context)
-        Features::ArgoSetup.new(git_template: git_template).call(directory_name, identifier, context)
+        dependencies = admin? ? [ArgoDependencies.node_installed(min_major: 16, min_minor: 1)] : []
```

Comment:
I don't think we can determine if they're running `npm` or `yarn` at extension creation, because the way we determine the package manager is by looking at the project file, [and seeing if the project has a `yarn.lock`](https://github.com/Shopify/shopify-app-cli/blob/master/lib/shopify-cli/js_system.rb#L65-L70), so if the project folder doesn't exist yet, we can't determine the package manager at that point in the flow. Unless I'm missing something :)

Reply:
> Node 16.0.0 gave me the same issues when I tested locally. Version 16.1.0 was the version that did not have any issues for me.

That's strange, I have node 16 which uses npm v7.10.0 and it works for me. Which version of npm is installed for you?

The release notes shows that it installs npm 7.10.0: https://medium.com/the-node-js-collection/node-js-16-available-now-7f5099a97e70

> the way we determine the package manager is by looking at the project folder, and seeing if the project has a yarn.lock

I think this doesn't work well for extensions as we have template repos that are not supposed to be opinionated about whether you use `yarn` or `npm` so in theory they could have both a yarn.lock and a package-lock.json so you could use either. In the future we might be getting rid of the lock files as well as it's often out of date and hard to maintain. Since this part of the code is argo specific, could we update to simply check whether yarn is globally installed and skip checking for the yarn.lock?

Reply:
I would pass the dependencies in from a higher level to avoid branching based on surface area here. These could come from the specification through the specification handler that is already responsible for instantiating this object.

Reply:
> That's strange, I have node 16 which uses npm v7.10.0 and it works for me. Which version of npm is installed for you?

@cecyc sorry I got mixed up - Node 16.0.0 does NOT work for the extraneous dependency issues. You're correct to set the min version to 16.1.0 👍